tests/unit/amo/components/TestPage.js

```diff
@@ -724,6 +726,209 @@ describe(__filename, () => {
+          utm_content: 'footer-link',
+          utm_campaign: null,
+        })}`,
+      );
+
+      expect(screen.getByRole('link', { name: 'Pocket' })).toHaveAttribute(
+        'href',
+        `https://getpocket.com${makeQueryStringWithUTM({
+          utm_content: 'footer-link',
+          utm_campaign: null,
+        })}`,
+      );
+
+      expect(screen.getByRole('link', { name: 'Desktop' })).toHaveAttribute(
+        'href',
+        `https://www.mozilla.org/firefox/new/${makeQueryStringWithUTM({
+          utm_content: 'footer-link',
+          utm_campaign: null,
+        })}`,
+      );
+
+      expect(screen.getByRole('link', { name: 'Mobile' })).toHaveAttribute(
+        'href',
+        `https://www.mozilla.org/firefox/mobile/${makeQueryStringWithUTM({
+          utm_content: 'footer-link',
+          utm_campaign: null,
+        })}`,
+      );
+
+      expect(screen.getByRole('link', { name: 'Enterprise' })).toHaveAttribute(
+        'href',
+        `https://www.mozilla.org/firefox/enterprise/${makeQueryStringWithUTM({
+          utm_content: 'footer-link',
+          utm_campaign: null,
+        })}`,
+      );
+
+      expect(
+        screen.getByRole('link', { name: 'Developer Policies' }),
+      ).toHaveAttribute(
+        'href',
+        `${extensionWorkshopUrl}/documentation/publish/add-on-policies/${makeQueryStringWithUTM(
+          {
+            utm_medium: 'photon-footer',
+            utm_campaign: null,
+          },
+        )}`,
+      );
+
+      expect(
+        screen.getByRole('link', { name: 'Community Blog' }),
+      ).toHaveAttribute(
+        'href',
+        `https://blog.mozilla.com/addons${makeQueryStringWithUTM({
+          utm_campaign: null,
+          utm_content: 'footer-link',
+          utm_medium: 'referral',
+        })}`,
+      );
+    });
+  });
+
+  describe('Tests for LanguagePicker', () => {
+    it('renders a LanguagePicker', () => {
+      render();
+
+      expect(
+        screen.getByRole('combobox', { name: 'Change language' }),
+      ).toBeInTheDocument();
+    });
+
+    it('selects the current locale', () => {
+      store = dispatchClientMetadata({ lang: 'fr' }).store;
+      render({ location: '/fr/firefox/' });
+
+      expect(screen.getByRole('option', { name: 'Français' }).selected).toEqual(
+        true,
+      );
+      expect(screen.getByRole('option', { name: 'Français' }).value).toEqual(
+        'fr',
+      );
+    });
+
+    it('changes the language on change', () => {
+      store = dispatchClientMetadata({ lang: 'fr' }).store;
+      render({ location: '/fr/firefox/' });
+      expect(window.location.pathname).toEqual('/fr/firefox/');
+
+      userEvent.selectOptions(
+        screen.getByRole('combobox', { name: 'Change language' }),
+        screen.getByRole('option', { name: 'Español' }),
+      );
+
+      expect(window.location).toEqual('/es/firefox/');
+    });
+
+    describe('changeLocaleURL', () => {
```

Comment:
Good idea. I did that in 1710ecbadfccf0582f20c105d2cfe61189bec948 ; I didn't keep the `describe` block, because the component already had a similar test.